File: packages/ios-sdk/Sources/PaaqIntelligence/PaaqConfig.swift
import Foundation

public struct PaaqConfig {
    public let sdkToken: String
    public let projectId: String
    public let environment: String
    public let debug: Bool
    public let flushInterval: TimeInterval
    public let heartbeatInterval: TimeInterval
    public let batchSize: Int

    public init(
        sdkToken: String,
        projectId: String,
        environment: String = "production",
        debug: Bool = false,
        flushInterval: TimeInterval = 30,
        heartbeatInterval: TimeInterval = 300,
        batchSize: Int = 50
    ) {
        self.sdkToken = sdkToken
        self.projectId = projectId
        self.environment = environment
        self.debug = debug
        self.flushInterval = flushInterval
        self.heartbeatInterval = heartbeatInterval
        self.batchSize = batchSize
    }
}

File: packages/android-sdk/src/main/kotlin/io/paaq/intelligence/PaaqConfig.kt
package io.paaq.intelligence

data class PaaqConfig(
    val sdkToken: String,
    val projectId: String,
    val environment: String = "production",
    val debug: Boolean = false,
    val flushIntervalSeconds: Long = 30L,
    val heartbeatIntervalSeconds: Long = 300L,
    val batchSize: Int = 50,
)
